pkg/retrieval/retrieval.go

```diff
@@ -93,14 +93,19 @@ func (s *Service) RetrieveChunk(ctx context.Context, addr swarm.Address) (data [
 				continue
 			}
 			s.logger.Tracef("retrieval: got chunk %s from peer %s", addr, peer)
-			return data, nil
+			//
+			chunk := swarm.NewChunk(addr, data)
+			if valid(chunk) {
+				return chunk, nil
+			}
```

Comment:
Code style:
I usually find it better to return the error-conditions inside the if-block, and do the happy-path outside if blocks.